syntax = "proto3";

import "google/protobuf/empty.proto";

option java_multiple_files = true;
option java_package = "io.grpc.examples.messenger";
option java_outer_classname = "MessengerProto";
option objc_class_prefix = "MSG";

package messenger;

service Messenger {
  rpc LogIn (LogInRequest) returns (google.protobuf.Empty) {}
  rpc OpenMessageStream (stream Message) returns (stream Message) {}
  rpc LogOut (LogOutRequest) returns (google.protobuf.Empty) {}
}

message LogInRequest {
  string login = 1;
  string password = 2;
}

message LogOutRequest {
  string login = 1;
}

message Credentials {
  string login = 1;
  string password = 2;
  uint64 id = 3;
}

message CredentialBook {
  repeated Credentials data = 1;
}

service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SayHelloStreamReply (HelloRequest) returns (stream HelloReply) {}
}

message Message {
  string sender = 1;
  string receiver = 2;
  string data = 3;
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}
